src/ripple/app/paths/PathRequest.cpp

```diff
@@ -585,10 +604,20 @@ PathRequest::findPaths (std::shared_ptr<RippleLineCache> const& cache, int const
         }
     }
 
+    // Charge cost is based on source currencies used.
+    double const min_cost = 50;
+    double const max_cost = 400;
+    Resource::Charge c {static_cast<int>(
+        std::min(max_cost, std::max(
+            min_cost, std::pow(sourceCurrencies.size(), 2)))),
+                "path update"};
+    consumer_.charge(c);
+
```

Comment:
I'd like to get rid of the `pow` and `static_cast` do the computations with `uint32`. Proposed (with `int` rather than `uint32` since `charge` takes an `int`:

```
    auto cost = [size = sourceCurrencies.size ()]
    {
        int const minCost = 50;
        int const maxCost = 400;
        static_assert(maxCost * maxCost < std::numeric_limits<int>::max(),"");
        if (size > maxCost)
            return maxCost;
        int const squared = size*size;
        return std::min(maxCost, std::max(minCost, squared));
    };
    Resource::Charge c {cost(), "path update"};
```